src/Nocarrier/Hal.php

```diff
@@ -240,7 +240,7 @@ public function setData(Array $data = null)
     public function getData($key = null)
     {
         if ($key) {
-            return isset($this->data[$key]) ? $this->data[$key] : null;
+            return isset($this->data[$key]) ? $this->data[$key] : array();
```

Comment:
Ah, yes, you are right of course. My bad.

I think it would be good to state in the docblock that it may return a scalar only when a key is given.